app/parse.py

```diff
@@ -1,18 +1,77 @@
 from dataclasses import dataclass
 from enum import Enum
 
+import requests
+from bs4 import BeautifulSoup
 
-class CourseType(Enum):
-    FULL_TIME = "full-time"
-    PART_TIME = "part-time"
+BASE_URL = "https://mate.academy/"
 
 
 @dataclass
 class Course:
     name: str
     short_description: str
-    course_type: CourseType
+    course_type: Enum
+
+
+class CourseType(Enum):
+    FULL_TIME = "full-time"
+    PART_TIME = "part-time"
+
+
+def parse_single_course(course_soup: BeautifulSoup) -> list[Course]:
+    name = course_soup.select_one(
+        ".typography_landingH3__vTjok.ProfessionCard_title__Zq5ZY.mb-12"
+    ).text
+    short_description = course_soup.select_one(
+        ".typography_landingTextMain__Rc8BD.mb-32"
+    ).get_text()
+
+    full_time_button = course_soup.select_one(
+        '[data-qa="fulltime-course-more-details-button"]'
+    )
+    part_time_button = course_soup.select_one(
+        '[data-qa="fx-course-details-button"]'
+    )
+
+    courses = []
+
+    if full_time_button and part_time_button:
+        full_time_course = Course(
+            name=name,
+            short_description=short_description,
+            course_type=CourseType.FULL_TIME)
+        part_time_course = Course(
+            name=name,
+            short_description=short_description,
+            course_type=CourseType.PART_TIME)
+        courses.append(full_time_course)
+        courses.append(part_time_course)
+    elif part_time_button:
+        part_time_course = Course(
+            name=name,
+            short_description=short_description,
+            course_type=CourseType.PART_TIME)
+        courses.append(part_time_course)
```

Comment:
what if only full_time? in this case it won't handle it

Reply:
fixed. With this code, all possible combinations of course availability are handled correctly.